src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1030,7 +1030,8 @@ VideoDecoder::DecodedOutput VideoDecoder::getFrameAtIndex(
 
 VideoDecoder::BatchDecodedOutput VideoDecoder::getFramesAtIndices(
     int streamIndex,
-    const std::vector<int64_t>& frameIndices) {
+    const std::vector<int64_t>& frameIndices,
+    const bool sortIndices) {
```

Comment:
The default of `sortIndices` is False right now. But I think it should be True. The benchmark shows that sorting an already-sorted frameIndices array is basically zero cost (within noise):

```
This PR:
ordered indices: True
med = 21.82ms +- 1.50 sort_indices = True
med = 21.71ms +- 1.38 sort_indices = False
```

So there seem to be no downside to always sorting by default.

OTOH, if we keep the default to False, some users might be missing out on a 10X speed-up, and that sounds overall worse.

Reply:
Agreed on decision and reasoning.

Reply:
IMHO we should just call std::is_sorted before sorting them (if they are found to be not sorted). I cannot imagine this parameter being useful to callers.

Reply:
I guess it's fine not to expose the `sortIndices` parameter at all.
But then, do we need to check for `is_sorted()` before we sort? Should we just always sort, and assume that the underlying implement is such that sorting an already-sorted array is O(n) anyway?

Reply:
We discussed offline, let's just do what Ahmad suggested.